### PR TITLE
add the materialize gtm script that is used on the website.

### PR DESCRIPTION
**Pull request by nisarhassan12**
Addresses https://materializeinc.slack.com/archives/C07MSU999NK/p1729112546624819?thread_ts=1729112347.720909&cid=C07MSU999NK

### CODE REVIEW

File: doc/user/layouts/partials/head.html [hunk 1]
```diff
@@ -93,57 +93,19 @@
...
-    {{/* Use cookies for storage by default, since they cross subdomains */}}
-    analytics.load(SEGMENT_ID, { storage: { stores: ["cookie", "localStorage", "memory"] } });
-  {{ else }}
-    {{/* Dev */}}
-    analytics._writeKey="dGeQYRjmGVsqDI0KIARrAhTvk1BdJJhk";
-    analytics.load("dGeQYRjmGVsqDI0KIARrAhTvk1BdJJhk", { storage: { stores: ["cookie", "localStorage", "memory"] } });
-  {{ end }}
-  analytics.page();
-  }}();
+  (function (w, d, s, l, i) {
```
Comment (matthelm) [inline doc/user/layouts/partials/head.html:97]: I notice that the existing code has `{{ if hugo.IsProduction }}`. I wonder if we also want that here?

**Review by matthelm — APPROVED** [review]
Looks good to me. Let's make sure before merging that @kay-kim and @RobinClowers don't have any objections.

File: doc/user/layouts/partials/head.html [hunk 2]
```diff
@@ -93,57 +93,19 @@
 </script>
 
 {{if hugo.IsProduction}} {{/* Google Analytics */}}
-<script async src="https://www.googletagmanager.com/gtag/js?id=G-HYXTGZN2KT"></script>
 <script>
-  window.dataLayer = window.dataLayer || [];
-  function gtag(){dataLayer.push(arguments);}
-  gtag('js', new Date());
-
-  gtag('config', 'G-HYXTGZN2KT');
```
Comment (matthelm) [inline doc/user/layouts/partials/head.html:102]: Confirming that this is the same ID we use within our Google Tag Manager container
image

File: doc/user/layouts/partials/head.html (hunk 1 above)
Comment (nisarhassan12) [reply doc/user/layouts/partials/head.html:97]: Sorry I'm unable to understand what you meant here that is still there wrapping the script tag.

**Review by kay-kim — APPROVED** [review]
no objections on my side.

**Review by matthelm — APPROVED** [review]

**Pull request merged**